x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -181,7 +181,8 @@
         this.description = description;
         this.tags = Collections.unmodifiableList(ExceptionsHelper.requireNonNull(tags, TAGS));
         this.metadata = metadata == null ? null : Collections.unmodifiableMap(metadata);
-        this.input = ExceptionsHelper.requireNonNull(input, INPUT);
+        this.input = (input == null && modelType == TrainedModelType.PYTORCH) ?
```

Comment:
++ A user should not have to think about which possible field will be used for input when defining the model config, it is easier to define at the point of use (e.g. in an ingest processor) when the field names are known.